src/fields/eval_points.cc

```diff
@@ -0,0 +1,123 @@
+/*!
+ *
+﻿ * Copyright (C) 2015 Technical University of Liberec.  All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 3 as published by the
+ * Free Software Foundation. (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ *
+ * @file    eval_points.cc
+ * @brief
+ * @author  David Flanderka
+ */
+
+#include "fields/eval_points.hh"
+#include "fields/eval_subset.hh"
+#include "quadrature/quadrature.hh"
+#include "mesh/ref_element.hh"
+#include <memory>
+
+
+const unsigned int EvalPoints::undefined_dim = 10;
+
+EvalPoints::EvalPoints()
+: local_points_(0, 1), dim_(EvalPoints::undefined_dim)
+{
+	subset_starts_.push_back(0);
+}
+
+template <unsigned int dim>
+EvalSubset EvalPoints::add_bulk(const Quadrature &quad)
+{
+	check_dim(quad.dim(), dim);
+
+	EvalSubset bulk_set(shared_from_this() );
+	this->add_local_points<dim>( quad.get_points() );
+	subset_starts_.push_back( this->size() );
+    return bulk_set;
+}
+
+template <unsigned int dim>
+EvalSubset EvalPoints::add_side(const Quadrature &quad)
+{
+	check_dim(quad.dim()+1, dim);
+	unsigned int old_data_size=this->size(), new_data_size; // interval of side subset data
+
+	EvalSubset side_set(shared_from_this(), true);
+	EvalSubset::PermutationIndices &perm_indices = side_set.perm_indices();
+
+    // permutation 0
+    for (unsigned int i=0; i<dim+1; ++i) {  // sides
+        Quadrature high_dim_q = quad.make_from_side<dim>(i, 0);
+        this->add_local_points<dim>( high_dim_q.get_points() );
+    }
+    new_data_size = this->size();
+    subset_starts_.push_back( new_data_size );
+    for (unsigned int i_perm=0, i_data=old_data_size; i_data<new_data_size; ++i_perm, ++i_data) {
+    	perm_indices[0][i_perm] = i_data;
+    }
+
+    // permutation 1...N
+    for (unsigned int j=1; j<RefElement<dim>::n_side_permutations; ++j) {
+        unsigned int i_perm=0;
+        for (unsigned int i=0; i<dim+1; ++i) {  // sides
+            Quadrature high_dim_q = quad.make_from_side<dim>(i, j);
+            const Armor::array & quad_points = high_dim_q.get_points();
+            for (uint k=0; k<quad_points.n_vals(); ++k) {
+            	perm_indices[j][i_perm] = this->find_permute_point<dim>( quad_points.get<dim>(k).arma(), old_data_size, new_data_size );
```

Comment:
Change `perm_indices` to be just a C array of indices. Use it through a method that computes pointer for 
i_side, i_perm, i_point. Usage:
``` 
eval.local_points(*perm_idx_ptr(i_side, i_perm, i_point))
```